Build fails with Go 1.23 because of changed testing.testDeps

When trying to build with Go 1.23 rc1 in
* https://github.com/Homebrew/homebrew-core/pull/175310

It failed at
https://github.com/hofstadter-io/hof/blob/bebc58f867e4341e3060f2421910acef6b743ea5/script/runtime/exe_next.go#L52

with [error](https://github.com/Homebrew/homebrew-core/actions/runs/9617673150/job/26535459459?pr=175310#step:3:10846):
````
  # github.com/hofstadter-io/hof/script/runtime
  script/runtime/exe_next.go:52:26: cannot use nopTestDeps{} (value of type nopTestDeps) as testing.testDeps value in argument to testing.MainStart: nopTestDeps does not implement testing.testDeps (missing method InitRuntimeCoverage)
````

`InitRuntimeCoverage` was added by @rsc in:
https://github.com/golang/go/commit/180ea45566d19e60aa2d660f6139b7f6e18ff56b#diff-36f401ff1ffadb7d7a39cf41c2329615b159fa95cf2e14c1f58d8c46331af429R208

See
* https://github.com/golang/go/issues/67401
* https://tip.golang.org/doc/go1.23#linker
* https://github.com/golang/go/blob/82c14346d89ec0eeca114f9ca0e88516b2cda454/src/testing/internal/testdeps/deps.go#L218-L223

Given the NOP nature of `nopTestDeps` a possible workaround could be to just implement a dummy/nop function `InitRuntimeCoverage()` in the `nopTestDeps`:
````go
func (nopTestDeps) InitRuntimeCoverage() (mode string, tearDown func(string, string) (string, error), snapcov func() float64) {
	return
}
````

--- a/flow/testdata/tasks/api/req_001.txt
+++ b/flow/testdata/tasks/api/req_001.txt
@@ -35,7 +35,12 @@ tasks: {
   mask: {
     @task(st.Mask)
     val: call.resp.body
-    mask: { headers: "x-amzn-trace-id": string }
+    mask: { 
+      headers: {
+        "x-amzn-trace-id": string
+        "x-request-start": string
+      }
+    }
   }
 	out: { text: json.Indent(json.Marshal(mask.out), "", "  ") +"\n" } @task(os.Stdout)
 }
@@ -46,9 +51,9 @@ tasks: {
     "cow": "moo"
   },
   "headers": {
+    "host": "postman-echo.com",
     "x-forwarded-proto": "https",
     "x-forwarded-port": "443",
-    "host": "postman-echo.com",
     "accept-encoding": "gzip",
     "user-agent": "Go-http-client/2.0"
   },

--- a/script/runtime/exe_next.go
+++ b/script/runtime/exe_next.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build go1.18
+//go:build go1.23
 
 package runtime
 
@@ -64,4 +64,8 @@ func runCoverSubcommand(cprof string, mainf func() int) (exitCode int) {
 		}
 	}()
 	return mainf()
+}
+
+func (nopTestDeps) InitRuntimeCoverage() (mode string, tearDown func(string, string) (string, error), snapcov func() float64) {
+	return
 }

--- a/script/runtime/exe_old.go
+++ b/script/runtime/exe_old.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !go1.18
+//go:build !go1.23
 
 package runtime
 
@@ -49,7 +49,7 @@ func runCoverSubcommand(cprof string, mainf func() int) (exitCode int) {
 		// Run MainStart (recursively, but it we should be ok) with no tests
 		// so that it writes the coverage profile.
 		// go1.18 -- m := testing.MainStart(nopTestDeps{}, nil, nil, nil, nil)
-		m := testing.MainStart(nopTestDeps{}, nil, nil, nil)
+		m := testing.MainStart(nopTestDeps{}, nil, nil, nil, nil)
 		if code := m.Run(); code != 0 && exitCode == 0 {
 			exitCode = code
 		}

